src/pretix/control/views/user.py

```diff
@@ -243,7 +244,7 @@ def form_valid(self, form):
         return sup
 
     def get_success_url(self):
-        return reverse('control:user.settings')
+        return self.request.GET.get('next', reverse('control:user.settings'))
```

Comment:
To prevent an [unvalidated redirect](https://cheatsheetseries.owasp.org/cheatsheets/Unvalidated_Redirects_and_Forwards_Cheat_Sheet.html), you should validate the URL first:

```suggestion
        if "next" in request.GET and url_has_allowed_host_and_scheme(request.GET.get("next"), allowed_hosts=None):
            return request.GET.get("next")
        return reverse('control:user.settings')
```

You'll also need

```
from django.utils.http import url_has_allowed_host_and_scheme
```